src/main/haskell/kore/src/Kore/Builtin/Builtin.hs

```diff
@@ -546,11 +544,25 @@ functionEvaluator ctx impl =
             , applicationChildren
             }
       =
-        Kore.Error.withContext ctx
-            (do
-                attempt <- impl tools simplifier resultSort applicationChildren
-                return (attempt, SimplificationProof)
-            )
-
-wrongArity :: MonadError (Error w) m => m a
-wrongArity = Kore.Error.koreFail "Wrong number of arguments"
+        do
+            attempt <- impl tools simplifier resultSort applicationChildren
+            return (attempt, SimplificationProof)
+
+{- | Evaluation failure due to a builtin call with the wrong arity.
+
+  Such a failure indicates a bug in the well-formedness checker.
+
+ -}
+wrongArity :: HasCallStack => String -> a
+wrongArity ctx = error (ctx ++ ": Wrong number of arguments")
+
+{- | Evaluation failure due to an unparseable domain value.
+
+  Such a failure indicates a bug in the well-formedness checker.
+
+ -}
+liftParser :: HasCallStack => String -> Either (Error e) a -> a
```

Comment:
You can leave it as-is, but consider naming this `extractParsedValue` or something similar.